### ISSUE TITLE
False-positive OutdatedDocumentation for generic parameters

### ISSUE THREAD

**Issue by mpetuska** [detekt/detekt#4727]
## Expected Behavior

`@param T` supported for generic type documentation

## Observed Behavior

`@param T` is reported as a false positive
`Documentation of JsonObject is outdated [OutdatedDocumentation]`

## Steps to Reproduce

Run detekt with `OutdatedDocumentation` rule enabled on this code
```kotlin
/**
 * Some interesting description
 * @param T value type
 */
public abstract class Test<T : Any>
```

## Your Environment

* Version of detekt used: 1.20.0
* Version of Gradle used (if applicable): 7.4
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: Linux/Fedora 35
* Link to your project (if it's a public repository): https://github.com/mpetuska/npm-publish

**Comment by OksiBlack** [comment]
The problem reappered 
Detekt 1.23.0
